File: testers/grpc/scala/generated-and-checked-in/com/example/grpc/OrderService.scala
package com.example.grpc



/** Clean service interface for OrderService gRPC service */
trait OrderService {
  def getCustomer(request: GetCustomerRequest): GetCustomerResponse

  def createOrder(request: CreateOrderRequest): CreateOrderResponse

  def listOrders(request: ListOrdersRequest): java.util.Iterator[OrderUpdate]

  def submitOrders(requests: java.util.Iterator[CreateOrderRequest]): OrderSummary

  def chat(requests: java.util.Iterator[ChatMessage]): java.util.Iterator[ChatMessage]
}

File: testers/grpc/kotlin/generated-and-checked-in/com/example/grpc/OrderService.kt
package com.example.grpc

import kotlin.collections.Iterator

/** Clean service interface for OrderService gRPC service */
interface OrderService {
  abstract fun chat(requests: Iterator<ChatMessage>): Iterator<ChatMessage>

  abstract fun createOrder(request: CreateOrderRequest): CreateOrderResponse

  abstract fun getCustomer(request: GetCustomerRequest): GetCustomerResponse

  abstract fun listOrders(request: ListOrdersRequest): Iterator<OrderUpdate>

  abstract fun submitOrders(requests: Iterator<CreateOrderRequest>): OrderSummary
}